core/store.go

```diff
@@ -242,6 +242,26 @@ func Get(k string) *Obj {
 	return getHelper(k, true)
 }
 
+func GetDel(k string) *Obj {
+	var v *Obj
+	withLocks(func() {
+		ptr, ok := keypool[k]
+		if !ok {
+			return
+		}
+
+		v = store[ptr]
+		if v != nil {
+			expired := hasExpired(v)
+			deleteKeyWithOperationName(k, ptr, v, "GETDEL")
+			if expired {
+				v = nil
+			}
+		}
+	}, WithStoreRLock(), WithKeypoolRLock())
```

Comment:
Since there is a possibility of a write operation (Deletion), we should be taking Write locks instead of Read locks.